controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -146,8 +146,16 @@
         Once.after(Duration.ofMinutes(1), () -> {
             Instant start = clock.instant();
             int count = 0;
-            for (Application application : curator.readApplications()) {
-                lockApplicationIfPresent(application.id(), this::store);
+            for (TenantAndApplicationId id: curator.readApplicationIds()) {
+                lockApplicationIfPresent(id, application -> {
+                    if (id.tenant().value().startsWith("by-"))
```

Comment:
In the future we should disallow creation of by tenants altogether and remove this.

Reply:
Absolutely.